compiler/rustc_typeck/src/check/expr.rs

```diff
@@ -2257,6 +2258,7 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         candidate_field: &ty::FieldDef,
         subst: SubstsRef<'tcx>,
         mut field_path: Vec<Ident>,
+        id: HirId,
```

Comment:
Rather than passing the `HirId` here, I think it probably makes more sense to directly pass through the `DefId` of the module, and produce this `DefId` back where `no_such_field_err` is called.

Reply:
Done.